RFS/src/main/java/com/rfs/version_os_2_11/GlobalMetadataCreator_OS_2_11.java

```diff
@@ -21,40 +25,29 @@ public class GlobalMetadataCreator_OS_2_11 {
     private final List<String> indexTemplateAllowlist;
     private final IMetadataMigrationContexts.IClusterMetadataContext context;
 
-    public GlobalMetadataCreator_OS_2_11(
-        OpenSearchClient client,
-        List<String> legacyTemplateAllowlist,
-        List<String> componentTemplateAllowlist,
-        List<String> indexTemplateAllowlist,
-        IMetadataMigrationContexts.IClusterMetadataContext context
-    ) {
-        this.client = client;
-        this.legacyTemplateAllowlist = legacyTemplateAllowlist;
-        this.componentTemplateAllowlist = componentTemplateAllowlist;
-        this.indexTemplateAllowlist = indexTemplateAllowlist;
-        this.context = context;
-    }
-
-    public void create(GlobalMetadata root) {
+    public GlobalMetadataCreatorResults create(GlobalMetadata root) {
```

Comment:
This classes is missing abstractions around the different templates - it is expecting OS v2.11 formatted global metadata snapshot format, this is cause for the client code around indices to be so messy.